BUG: Error while converting Kraken2 reports (on dereplicated MAGs) to taxonomy

**Describe the bug**
Dear devs,
I am facing an error while trying to assign taxonomies to dereplicated MAGs.
Background:
- Reads after host DNA removal and QC (outside q2)
- Assemblies with MetaSPAdes  (outside q2)
- Binning with COMEbin  (outside q2)
- Quality filtering of bins  (outside q2)
- Dereplication (within q2)
- Taxonomy annotation with kraken2  (witin q2)

Additional info:
- Kraken2 + bracken on cleaned reads worked without issues


**To Reproduce**
Qiime 2024.10

```
conda activate qiime2-metagenome-2024.10

#taxonomy with kraken2
qiime moshpit classify-kraken2 \
    --i-seqs $PS/MAGs/derep_mags.qza \
    --i-kraken2-db $DB/Kraken2_PlusPF/kraken2.qza \
    --p-threads {THREADS} \
    --p-confidence 0.5 \
    --p-report-minimizer-data \
    --o-reports ./cache:kraken_reports_mags_derep \
    --o-hits ./cache:kraken_hits_mags_derep \
    --verbose

#convert to q2 taxonomy
qiime moshpit kraken2-to-mag-features \
    --i-reports ./cache:kraken_reports_mags_derep  \
    --i-hits ./cache:kraken_hits_mags_derep  \
    --o-taxonomy $PS/Taxonomy/taxonomy_mags.qza \
    --verbose
```
Kraken2 creates reports and hits.
Error while converting:

```
/home/ho/ho_genetics/ho_graaf20/miniforge3/envs/qiime2-metagenome-2024.10/lib/python3.10/site-packages/q2_moshpit/kraken2/select.py:164: FutureWarning: Downcasting object dtype arrays on .fillna, .ffill, .bfill is deprecated and will change in a future version. Call result.infer_objects(copy=False) instead. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`
  table = pd.DataFrame(rows).fillna(False)
Traceback (most recent call last):
  File "/home/ho/ho_genetics/ho_graaf20/miniforge3/envs/qiime2-metagenome-2024.10/lib/python3.10/site-packages/q2cli/commands.py", line 530, in __call__
    results = self._execute_action(
  File "/home/ho/ho_genetics/ho_graaf20/miniforge3/envs/qiime2-metagenome-2024.10/lib/python3.10/site-packages/q2cli/commands.py", line 602, in _execute_action
    results = action(**arguments)
  File "<decorator-gen-713>", line 2, in kraken2_to_mag_features
  File "/home/ho/ho_genetics/ho_graaf20/miniforge3/envs/qiime2-metagenome-2024.10/lib/python3.10/site-packages/qiime2/sdk/action.py", line 299, in bound_callable
    outputs = self._callable_executor_(
  File "/home/ho/ho_genetics/ho_graaf20/miniforge3/envs/qiime2-metagenome-2024.10/lib/python3.10/site-packages/qiime2/sdk/action.py", line 570, in _callable_executor_
    output_views = self._callable(**view_args)
  File "/home/ho/ho_genetics/ho_graaf20/miniforge3/envs/qiime2-metagenome-2024.10/lib/python3.10/site-packages/q2_moshpit/kraken2/select.py", line 116, in kraken2_to_mag_features
    table, taxonomy = kraken2_to_features(reports, coverage_threshold)
  File "/home/ho/ho_genetics/ho_graaf20/miniforge3/envs/qiime2-metagenome-2024.10/lib/python3.10/site-packages/q2_moshpit/kraken2/select.py", line 168, in kraken2_to_features
    taxonomy = taxonomy.loc[list(table.columns)]
  File "/home/ho/ho_genetics/ho_graaf20/miniforge3/envs/qiime2-metagenome-2024.10/lib/python3.10/site-packages/pandas/core/indexing.py", line 1191, in __getitem__
    return self._getitem_axis(maybe_callable, axis=axis)
  File "/home/ho/ho_genetics/ho_graaf20/miniforge3/envs/qiime2-metagenome-2024.10/lib/python3.10/site-packages/pandas/core/indexing.py", line 1420, in _getitem_axis
    return self._getitem_iterable(key, axis=axis)
  File "/home/ho/ho_genetics/ho_graaf20/miniforge3/envs/qiime2-metagenome-2024.10/lib/python3.10/site-packages/pandas/core/indexing.py", line 1360, in _getitem_iterable
    keyarr, indexer = self._get_listlike_indexer(key, axis)
  File "/home/ho/ho_genetics/ho_graaf20/miniforge3/envs/qiime2-metagenome-2024.10/lib/python3.10/site-packages/pandas/core/indexing.py", line 1558, in _get_listlike_indexer
    keyarr, indexer = ax._get_indexer_strict(key, axis_name)
  File "/home/ho/ho_genetics/ho_graaf20/miniforge3/envs/qiime2-metagenome-2024.10/lib/python3.10/site-packages/pandas/core/indexes/base.py", line 6200, in _get_indexer_strict
    self._raise_if_missing(keyarr, indexer, axis_name)
  File "/home/ho/ho_genetics/ho_graaf20/miniforge3/envs/qiime2-metagenome-2024.10/lib/python3.10/site-packages/pandas/core/indexes/base.py", line 6252, in _raise_if_missing
    raise KeyError(f"{not_found} not in index")
KeyError: "['r1__cellular organisms'] not in index"

Plugin error from moshpit:

  "['r1__cellular organisms'] not in index"

See above for debug info.
```

Qiime2 2025.4 

```
conda activate qiime2-moshpit-2025.4

#taxonomy with kraken2
qiime annotate classify-kraken2 \
    --i-seqs $PS/MAGs/derep_mags.qza \
    --i-kraken2-db $DB/Kraken2_PlusPF/kraken2.qza \
    --p-threads {THREADS} \
    --p-confidence 0.5 \
    --p-report-minimizer-data \
    --o-reports ./cache:kraken_reports_mags_derep \
    --o-hits ./cache:kraken_hits_mags_derep \
    --verbose
    
#convert to q2 taxonomy
qiime annotate kraken2-to-mag-features \
    --i-reports ./cache:kraken_reports_mags_derep  \
    --i-hits ./cache:kraken_hits_mags_derep  \
    --o-taxonomy $PS/Taxonomy/taxonomy_mags.qza \
    --verbose
```
Kraken2 creates reports and hits.
Error while converting:

```
Traceback (most recent call last):
  File "/home/ho/ho_genetics/ho_graaf20/miniforge3/envs/qiime2-moshpit-2025.4/lib/python3.10/site-packages/q2cli/commands.py", line 529, in __call__
    results = self._execute_action(
  File "/home/ho/ho_genetics/ho_graaf20/miniforge3/envs/qiime2-moshpit-2025.4/lib/python3.10/site-packages/q2cli/commands.py", line 601, in _execute_action
    results = action(**arguments)
  File "<decorator-gen-956>", line 2, in kraken2_to_mag_features
  File "/home/ho/ho_genetics/ho_graaf20/miniforge3/envs/qiime2-moshpit-2025.4/lib/python3.10/site-packages/qiime2/sdk/action.py", line 221, in bound_callable
    outputs = self._callable_executor_(
  File "/home/ho/ho_genetics/ho_graaf20/miniforge3/envs/qiime2-moshpit-2025.4/lib/python3.10/site-packages/qiime2/sdk/action.py", line 359, in _callable_executor_
    output_views = self._callable(**view_args)
  File "/home/ho/ho_genetics/ho_graaf20/miniforge3/envs/qiime2-moshpit-2025.4/lib/python3.10/site-packages/q2_annotate/kraken2/select.py", line 116, in kraken2_to_mag_features
    table, taxonomy = kraken2_to_features(reports, coverage_threshold)
  File "/home/ho/ho_genetics/ho_graaf20/miniforge3/envs/qiime2-moshpit-2025.4/lib/python3.10/site-packages/q2_annotate/kraken2/select.py", line 162, in kraken2_to_features
    full_tree = _combine_ncbi_trees(trees)
  File "/home/ho/ho_genetics/ho_graaf20/miniforge3/envs/qiime2-moshpit-2025.4/lib/python3.10/site-packages/q2_annotate/kraken2/select.py", line 268, in _combine_ncbi_trees
    raise AssertionError(f"{tip.name} could not be inserted")
AssertionError: g1__unclassified Ruminococcus could not be inserted

Plugin error from annotate:

  g1__unclassified Ruminococcus could not be inserted

See above for debug info.
```

**Expected behavior**
Conversion of reports to qiime2 taxonomy format

**Please complete the following information:**
 - OS: Rocky Linux 9.4 + SLURM (HPC)

**Additional context**
I can parse the reports that I have directly to get the taxonomies, but I guess that the bug should be reported here.

Best,
Timur

--- a/q2_annotate/kraken2/select.py
+++ b/q2_annotate/kraken2/select.py
@@ -312,9 +312,11 @@ def _combine_ncbi_trees(trees):
                         assert tip.name in tip_cache
                         subtree_inserted = True
 
-                if not subtree_inserted:
+                # did we exhaust our parents and still not find ourselves?
+                if not subtree_inserted and not (tip.name == matching.name):
                     # should be impossible. Implies ancestor_found was always
-                    # True but not in tip_cache
+                    # True but not in tip_cache AND the tip is not an
+                    # intermediate node in the tree that already is present.
                     raise AssertionError(f"{tip.name} could not be inserted")
     return full_tree
 

